MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m

```diff
@@ -85,14 +107,39 @@ + (instancetype)sharedInstance {
 - (void)startWithLogManager:(id<MSLogManager>)logManager appSecret:(NSString *)appSecret {
   [super startWithLogManager:logManager appSecret:appSecret];
 
+  // TODO: Hook up with pipeline.
+  NSURL *url;
+  MSLogInfo([MSUpdates logTag], @"Request updates token.");
+
+  // Most failures here require an app update. Thus, it will be retried only on next App instance.
+  @try {
+    url = [self buildTokenRequestURLWithAppSecret:appSecret];
+
+    // iOS 9+ only, check for SFSafariViewController availability.
+    Class clazz = NSClassFromString(@"SFSafariViewController");
+    if (clazz) {
+      
+      // Manipulate App UI on the main queue.
+      dispatch_async(dispatch_get_main_queue(), ^{
+        [self openURLInEmbeddedSafari:url fromClass:clazz];
+      });
+    } else {
+
+      // iOS 8.x.
+      [self openURLInSafariApp:url];
+    }
+  } @catch (NSException *exception) {
+    MSLogError([MSUpdates logTag], @"%@", exception.reason);
```

Comment:
Add explanation of what happened here like `@"Some more details: %@", exception.reason`?